xcube/core/resampling/rectify.py

```diff
@@ -602,22 +602,30 @@ def _compute_var_image_xarray_numpy(
 
 def _compute_var_image_xarray_dask(
     src_var: xr.DataArray,
-    dst_src_ij_images: np.ndarray,
+    dst_src_ij_images: da.Array,
     fill_value: Union[int, float, complex] = np.nan,
     interpolation: int = 0,
 ) -> da.Array:
     """Extract source pixels from xarray.DataArray source
     with dask.array.Array data.
     """
-    return da.map_blocks(
+    if src_var.ndim == 2:
+        src_var = src_var.expand_dims(dim={"dummy": 1})
```

Comment:
Please add comment explaining why this is needed. Questions a reader has here: Why expand 2 to 3 dims? Why isn't that asserted before calling this function? Why is 1 dim not handled?